docs/shared/deprecated/name-and-directory-format.md

```diff
@@ -0,0 +1,56 @@
+# nameAndDirectoryFormat
+
+As of Nx 17, many generators will prompt you to choose how Nx will calculate where the code should be located. You can choose between `as-provided` and `derived`. `as-provided` will be the default in Nx 18. The following flags are deprecated and will be removed in Nx 18: `--project`, `--flat`, `--pascalCaseFiles`, `--pascalCaseDirectory`, `--fileName`.
+
+## `as-provided`
+
+This setting makes generators behave in the following way:
+
+- `nx g component my-component` creates a component in the root.
+- `nx g component my-component --directory=apps/nested/my-app` creates a new component named `my-component` in the `/apps/nested/my-app` folder
+- `nx g component apps/nested/my-app/my-component` creates a new component named `my-component` in the `/apps/nested/my-app` folder
+
+{% callout type="note" title="Must generate inside a project" %}
+If the directory specified is not inside a project, an error will be thrown.
+{% /callout %}
+
+## `derived`
```

Comment:
I would definitely mention that this is for "backwards compatibility"

The story should be along the lines of...

If you are used to the old way, and you revert to old habits of not `cd`ing or not specifying the directory, this option is there to catch you. But be reminded that this option will be going away in v18, so don't rely on it. We also show the new command that people should be running so I would mention that here too.

```
✔ What should be the project name and where should it be generated? · lib2 @ libs/lib2
In Nx 18, generating projects will no longer derive the name and root.
Please provide the exact project name and root in the future.
Example: nx g @nx/vue:library lib2 --directory lib2
```

Reply:
I'll make it more prominent that this is the deprecated way of doing it